brave/brave-core
# Fix UI initialization flow and reset when devices < 2 from native side

fix https://github.com/brave/brave-browser/issues/2533
fix https://github.com/brave/brave-browser/issues/2537
fix https://github.com/brave/brave-browser/issues/2540
fix https://github.com/brave/brave-browser/issues/2552
fix https://github.com/brave/brave-browser/issues/2575

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
Except test case in above issues, here is the critical case:
1. Create sync chain on device a
2. Connect to sync chain on device b
3. Remember the sync code of the sync chain
4. Delete device b on device a
5. Both device should be reset to unsync state after a while
6.  Create sync chain on device a, the sync code should be different than step 3
7.  Create sync chain on device b, the sync code should be different than step 3

## Reviewer Checklist:
